Modules/_testcapimodule.c

```diff
@@ -4283,6 +4338,15 @@ PyInit__testcapi(void)
                            (PyObject *) &ContainerNoGC_type) < 0)
         return NULL;
 
+    PyObject *manual_heap_type = create_manual_heap_type();
+    if (manual_heap_type == NULL) {
+        return NULL;
+    }
+    if (PyModule_AddObject(m, "ManualHeapType", manual_heap_type) < 0) {
+        return NULL;
```

Comment:
I think we need to decref `manual_heap_type` on error, otherwise [it leaks](https://docs.python.org/3/c-api/module.html#c.PyModule_AddObject). Alternatively, we could use `PyModule_Add()`.